client_test.go

```diff
@@ -320,12 +320,12 @@ func TestProtectUnprotectCommandsPubKey(t *testing.T) {
 	}
 
 	command := []byte{0x05}
-	sharedKey, err := curve25519.X25519(c2PrivateCurveKey, e4crypto.PublicEd25519KeyToCurve25519(clientEdPk))
+	sharedKeyKey, err := curve25519.X25519(c2PrivateCurveKey, e4crypto.PublicEd25519KeyToCurve25519(clientEdPk))
```

Comment:
`sharedKeyKey` seems a bit of a redundant name :) If an intermediate name is needed (because the shared key is derived using sha3) then you could consider `sharedPoint` as that's what it is, an elliptic curve point that both parties now know.

Reply:
Good catch, fixed!